### DISCUSSION TITLE
How to prevent user from changing some part of code

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21750] [New to GitHub]
<p>Hello everyone,</p>
<p>I’ve made an offline windows application which user can donate in the about widnow. It would appear everybody can change donation adress to his. How can i prevent user to do such a thing. I don’t have no problem with changing other parts. </p>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/kokabi1365">@kokabi1365</a> ,</p>
<p>I’d recommend not to allow others to directly push to your repository.</p>
<p>Instead, let them create their own fork (that is, a copy of your repo on their account) and then a pull request on your repo. You, or the moderators you can optionally appoint, will then have a chance to review every change.</p>
<p>Hope that helps!</p>

**Comment by ghost** [comment]
<p>Okay I know, but contributer can copy my code to another website like bitbucket with his/her ownership or worse he/she can keep licence and publish software as a comerical closed source!</p>

**Comment by mpboom** [comment, marked as answer]
<p><em>I am not a lawyer, this is not legal advice. If you’d like legal advice consult an acutal lawyer - never accept legal advice from online strangers.</em></p>
<p>Straight out copying your sorce code to another location without the permission from either you or a license would be considered copyright enfrigement. See <a href="https://opensource.guide/legal/" rel="nofollow noopener">https://opensource.guide/legal/</a> for more information.</p>
<p>So technically you can never prevent this since, well, your code is open source. The only way to prevent it would be to keep it closed source. However, you can - using the correct license - cover this problem up in a “legal” way so you can take action if someone steals your code or tries to impersonate you or your project.</p>
